### DISCUSSION TITLE
HOGWARTS: Getting Expel button to delete

### DISCUSSION THREAD

**Discussion by TaylorHardaway** [nss-evening-cohort-14/evening-client-side#74] [Help Me!] [answered]
Taylor Hardaway
Yes, there is one similar but it is ahead of me. I have an Expel button on the card and a function called expelButtonEvents.
const expelButtonEvents = () => {
  document.querySelector('#Expel').addEventListener('click', expelButton);
}
I also wrote in the Expel id in the div code for the button
and my expelButton function looks like this
const expelButton = (e) => {
  const targetType = e.target.type;
  const targetId = e.target.id;
  if (targetType === 'button') {
    students.splice(targetId, 1);
  } 
  I know I need to put some kind of id's on the cards, but I thought I would have some success with this, but it does nothing.

Some sites I have googled
https://stackoverflow.com/questions/59776630/remove-card-when-clicked-on-confirm-button
http://www.randomsnippets.com/2008/03/26/how-to-dynamically-remove-delete-elements-via-javascript/
https://www.codecademy.com/forum_questions/4fc4e20c48eae700030456bb

Mostly I have fixing syntax errors, and I thought this would at least do something.
https://github.com/TaylorHardaway/sorting-hat

**Comment by TrinityChristiana** [comment]
Can you add what we did to fix this to the ticket?

**Comment by TaylorHardaway** [comment, marked as answer]
So I needed to pass #students as the argument for the querySelector to target the card that the students are in rather than an Id for the button.

**Reply by jimguistical** [reply]
Thank you, Taylor - this was super helpful!
